### ISSUE TITLE
Unhandled Exception: type 'Null' is not a subtype of type 'List<Object?>'

### ISSUE THREAD

**Issue by masterix21** [transistorsoft/flutter_background_geolocation#1147]
Hi. 

Using the last version I read the exception:
```
E/flutter (10830): [ERROR:flutter/runtime/dart_vm_initializer.cc(41)] Unhandled Exception: type 'Null' is not a subtype of type 'List<Object?>'
E/flutter (10830): #0      BackgroundGeolocation.getGeofence (package:flutter_background_geolocation/models/background_geolocation.dart:836:21)
```

I think you should handle better the null value here:
https://github.com/transistorsoft/flutter_background_geolocation/blob/a05ee82ef66dac60afb251900846ed10d9105aef/lib/models/background_geolocation.dart#L836C7-L836C44

**Comment by christocracy** [comment]
How do I reproduce this?

**Comment by masterix21** [comment]
It's hard to know.. but, I add a geofence like this:
```dart
await bg.BackgroundGeolocation.addGeofence(bg.Geofence(
    identifier: company.id.toString(),
    radius: radius ?? ENV.VALUABLE_PRESENCE_METERS,
    latitude: company.address!.position.latitude,
    longitude: company.address!.position.longitude,
    notifyOnEntry: true,
    notifyOnExit: true,
    notifyOnDwell: true,
    loiteringDelay: 3 * 1000,
    extras: {
        'addressId': company.address!.id,
        'companyId': company.id,
    }
));
```

The line that generates the exception is:
```dart
if (await bg.BackgroundGeolocation.getGeofence(company.id.toString()) != null) {
    return true;
}
```

**Comment by christocracy** [comment]
Well, I'd say that your `company.id.toString()` is returning `null` for you at some point.

**Comment by christocracy** [comment]
You ignored the issue template do I have no idea what version this comes from.




## Your Environment
* Plugin version:
* Platform: iOS or Android
* OS version:
* Device manufacturer / model:
* Flutter info (`flutter doctor`):
* Plugin config:
```dart <-- Syntax highlighting: DO NOT REMOVE -->
PASTE_YOUR_CODE_HERE
```

## Expected Behavior


## Actual Behavior


## Steps to Reproduce

1.
2.
3.
4.

## Context


## Debug logs

<details>
	<summary>Logs</summary>

``` <!-- syntax-highligting:  DO NOT REMOVE -->
PASTE_YOUR_LOGS_HERE
```

</details>

**Comment by christocracy** [comment]
Nor the platform which generated this (iOS vs Android)

**Comment by christocracy** [comment]
I will assume this was `v4.13.1` on *Android*.

Test fix by installing from `Git`:

```yaml
dependencies:
  flutter_background_geolocation:
    git:
      url: https://github.com/transistorsoft/flutter_background_geolocation.git
      ref: release-4.13.2
```

**Comment by masterix21** [comment]
I have the same issue with iOS and Android. 

> Well, I'd say that your `company.id.toString()` is returning `null` for you at some point.

Can't be the company because it can't be null when I add a geofence (I have a `if (company == null)` before).

I'll try your release-4.13.2.

**Comment by christocracy** [comment]
it was not your `company.id.toString()`.  It was a bug in the native side not sending a new attribute `Geofence.vertices` in the native `Map` in support of the new [*Polygon Geofencing*](https://transistorsoft.medium.com/polygon-geofencing-d99169a90538) feature just released.

**Comment by masterix21** [comment]
Ok. Finally, I have a question.

What's the right way to use your package in geofence mode only when the app is not in the foreground?

**Comment by christocracy** [comment]
Calling `.startGeofences()` places the plug-in to geofencing-only mode.

**Comment by masterix21** [comment]
Using startGeofences() I can't track the real-time position in the foreground mode. Right?

**Comment by christocracy** [comment]
.startGeofences() = track geofences **only**.

.start() = track geofences + location.

Simple.

**Comment by masterix21** [comment]
Sure, simple, but exists a parameter to stop tracking location when the app goes in background mode using only start()?

Foreground = track geofences + location

Background = track geofences only

**Comment by christocracy** [comment]
- .stop() = stop everything
- .start() = track location + geofences
- .startGeofences() = track only geofences.

I don't understand your confusion.

**Comment by masterix21** [comment]
In my app, if you are in the foreground, you’ll see a map with your real-time position, but if your app is in the background (or closed), the app only works in geofence mode.

Using “start()” only, the app still tracks the user’s real-time position when it is closed or in background mode.

**Comment by christocracy** [comment]
You know there's a method called .getCurrentPosition(), right?  Start a Timer and call it.

When app suspends, stop your timer.

**Comment by masterix21** [comment]
Issue solved with release-4.13.2. Thanks for your time.

**Comment by christocracy** [comment]
The plugin automatically uses infinite-geofencing when the number of geofences you add is greater than the platform maximum.
